packages/SwingSet/src/supervisors/supervisor-helper.js

```diff
@@ -120,12 +120,11 @@ function makeSupervisorSyscall(syscallToManager, workerCanBlock) {
     },
     vatstoreSet: (key, value) => doSyscall(['vatstoreSet', key, value]),
     vatstoreGetAfter: (priorKey, lowerBound, upperBound) => {
-      const result = doSyscall([
-        'vatstoreGetAfter',
-        priorKey,
-        lowerBound,
-        upperBound,
-      ]);
+      const syscall = ['vatstoreGetAfter', priorKey, lowerBound];
+      if (upperBound) {
+        syscall.push(upperBound);
```

Comment:
hm, I can imagine what prompted this, but I'd prefer to avoid variable-length syscall objects (e.g. VatSyscallVatstoreGetAfter` is always a length-4 array, with the last property being of variable type, rather than sometimes it's length-3 and sometimes it's length-4). Was the problem resolved by the `djson` fix you made? Or could we resolve it by doing something like `if (!upperBound) { syscall[3] = null; }` or something?

I know we have a general problem (#4390) with null-vs-undefined as we convert through both the kernel/worker pipe and to/from the transcript on the vat-manager side, and this local issue is a consequence of that technical debt.

Reply:
The problem with `dsjon` was something I tripped over but wasn't ultimately _the_ problem (though it was real).  Requiring the syscall array to be fixed length is very doable, but the practical effect would simply be to move this bit of conditional logic from the place the syscall array is constructed to the place where it is used, so I'm not sure that would be a win.